### ISSUE TITLE
Weighted version of the KLL sketch?

### ISSUE THREAD

**Issue by thvasilo** [apache/incubator-datasketches-cpp#157]
Hello,

There's a consideration at [XGBoost](https://github.com/dmlc/xgboost/issues/5746) about potentially using the KLL sketch to represent feature value histograms.

One potential blocker is the need for a weighted version of the sketch, this would allow us to use data points that are weighted, and adjust their feature contributions accordingly (See Appendix A of [XGBoost paper](https://arxiv.org/abs/1603.02754)).

I remember discussing in the past the possibility of using data weights with KLL, is that still an option?

**Comment by leerho** [comment]
The only option we have thought about would be restricted to positive integer weights and >= 1.

My interpretation of weights in a quantiles context is that an item with a weight of 2 is equivalent to updating the sketch with two identical items with a weight of 1.  Is this your understanding?

**Comment by thvasilo** [comment]
Yes, I think the outcome would be the same in this case.  I think for this to be used in XGBoost it would require real-valued weights.

I found the paper that I had in mind when talking to Zohar Karnin a couple of years ago, that includes a [weighted extension for KLL](https://arxiv.org/pdf/1907.00236.pdf) (Section 4).

I'll ping @trivialfis in case he wants to chime in.
